msteams-platform/bots/how-to/teams-conversational-ai/teams-conversation-ai-overview.md

```diff
@@ -141,21 +141,177 @@ The following table lists the updates to the Teams AI library:
 
+## Function calls using AI SDK
+
+Function calls, implemented within the AI SDK, unlock numerous capabilities, enabling the AI model to generate accurate responses seamlessly. It enables direct connection with external tools, thereby making AI even more powerful. These capabilities include performing complex calculations, retrieving important data, creating smoother workflows, and enabling dynamic interactions with users.
+
+> [!NOTE]
+> Structured outputs aren't supported.
+
+To use function calling with the Chat Completions API:
+
+1. Set up the planner where the default prompt uses the Tools Augmentation. Update one of the following files of your bot app:
+
+    * For a JavaScript app: Update `index.ts`.
+    * For a C# bot app: Update `Program.cs`.
+    * For a Python app: Update `bot.py`.
+
+   The following code snippet shows how to set up the `ToolsAugmentation` class:
+
+    # [JavaScript](#tab/javascript)
+
+    ```JavaScript
+    const planner = new ActionPlanner({
+        model,
+        prompts,
+        defaultPrompt: 'tools'
+    });
+    ```
+
+    # [C#](#tab/dotnet)
+
+    ```C#
+    ActionPlannerOptions<TurnState> options = new ActionPlannerOptions<TurnState>() 
+    { 
+        Model = model,
+        Prompts = prompts,
+        async (context, state, planner) =>
+        {
+            return await Task.FromResult(prompts.GetPrompt("Tools"));
+        } 
+    }
+    ActionPlanner<TurnState> planner = new ActionPlanner(options)
+    ```
+
+    # [Python](#tab/python)
+
+    ```Python
+    planner = ActionPlanner(ActionPlannerOptions(model=model, prompts=prompts, default_prompt="tools"))
+    ```
+
+    ---
+
+1. Specify tools augmentation in the `config.json` file.
+
+    ```JSON
+    {
+        "schema": 1.1,
+        "description": "",
+        "type": "",
+        "completion": {
+    +       "tool_choice": "auto",
+    +       "parallel_tool_calls": true,
+        },
+    +    "augmentation": {
+    +        "augmentation_type": "tools"
+    +    }
+    }
+    ```
+
+1. Specify all your `function definitions` in the `actions.json` file, which is in the `prompts` folder. Ensure that you follow the schema to avoid errors when the action is called by the LLM.
+
+    ```JSON
+    [{
+        "name": "CreateList",
+        "description": "Creates a list"
+    }]
+    ```
+
+1. Register your `handlers` in your `application` class.
+
+    * Each handler is a callback function that runs when a specific event happens. The function call handler executes code in response to the event.
+    * The function call must return a string as the output of the function call.
+    * After all functions are executed, the outputs are returned to the model. When the model requests to invoke any functions, these are mapped to `DO` commands within a `Plan` and are invoked in the AI class `run` function. The outputs are then returned to the model with tool call IDs to show that the tools were used.
```

Comment:
can remove this first sentence "After all functions are executed, the outputs are returned to the model."

Reply:
@lilyydu Updated input.